test: MMQA-224 add test eth to weth

racitores opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
Add tests based on PR https://github.com/MetaMask/metamask-extension/pull/30896
1. Add ETH to WETH swap 
2. Add send swap ETH to WETH (removed the previous one)

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
3. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31278?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
4.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

racitores committed:
test: add test eth to weth

racitores committed:
test: add send swap eth to weth

racitores committed:
fix: lint and add details

racitores committed:
refactor

racitores committed:
lint

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -75,124 +79,235 @@ async function mockSwapQuotes(mockServer: MockttpServer) {
-                ],
+  const swapTestCases = [
+    // {
+    //   name: 'should swap WETH to ETH',
+    //   sourceToken: 'WETH',
+    //   destinationToken: 'Ether',
+    //   sourceAmount: '10',
+    //   expectedWethBalance: '40',
+    //   expectedEthBalance: '34.99991',
+    //   dismissWarning: false,
+    // },
```
nit: do we need this commented code?

seaona commented on test/e2e/page-objects/pages/swap/swap-page.ts:
```diff
@@ -19,6 +19,11 @@ class SwapPage {
     text: 'Close',
   };
 
+  private readonly closeButtonId = '[data-testid="page-container-footer-next"]';
```
nit: 

```suggestion
  private readonly closeButton = '[data-testid="page-container-footer-next"]';
```

seaona replied:
side: is there a reason on why to use the id instead of the `closeButton` existing element?

I'm a bit doubtful on use this id, given that its used by many buttons in our confirmation pages, so maybe rather use the name as well? A combination of testId and text is also fine

racitores replied:
I was using the text but it seems something is failing, so just testing searching by id, etc... I will always prefer seach by text. Is this your favourite option too? And agree I'll remove this new locator I added 👍

seaona commented on test/e2e/page-objects/pages/swap/swap-page.ts:
```diff
@@ -19,6 +19,11 @@ class SwapPage {
 
+  private readonly closeButtonId = '[data-testid="page-container-footer-next"]';
+
+  private readonly transactionHeaderLocator =
```
nit
```suggestion
  private readonly transactionHeader =
```

seaona commented on test/e2e/page-objects/pages/swap/swap-page.ts:
```diff
@@ -54,10 +59,32 @@ class SwapPage {
     await this.driver.clickElement(this.swapButton);
-    await this.driver.clickElement(this.closeButton);
+    // console.log('Processing Swap');
+    // await this.swapProcessingMessageCheck('Processing');
```
could we remove this?

seaona commented on test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts:
```diff
@@ -0,0 +1,297 @@
+        async ({ driver }) => {
+          const swapSendPage = new SwapSendPage(driver);
+          await logInWithBalanceValidation(driver);
+
```
maybe we could check the initial token balances, so we make sure those are right at the beginning of the spec?

```suggestion
await homePage.check_expectedTokenBalanceIsDisplayed('50', 'WETH');
await homePage.check_expectedTokenBalanceIsDisplayed('25', 'ETH');

```

seaona replied:
this check will validate the 50ETH but not the 50 WETH. Could we add that too? I think it make sense for the test as we are swapping between the 2 assets, so we can check both balances before and after, if that makes sense?

await homePage.check_expectedTokenBalanceIsDisplayed('50', 'WETH');

seaona commented on test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts:
```diff
@@ -0,0 +1,297 @@
+          await homePage.goToTokensTab();
+          await homePage.check_expectedTokenBalanceIsDisplayed('60', 'WETH');
+          await homePage.check_expectedTokenBalanceIsDisplayed(
+            '14.99994',
```
the spec is failing as this value is incorrect

```suggestion
            '14.99995',
```

racitores replied:
I noticed that it is changing... For some reason is always this on my local and something else in the ci... I'll try with mocks

seaona replied:
mmm that's interesting :thinking: I could reproduce this consistently in my local. Might it be some rounding issue between OS's :thinking:

racitores replied:
yes, I think so... for now I have removed the check. 
I also tried to implement a function to pass the number of decimals to check, but the only way I was able to do it was with the anti-pattern const text = await asset.getText();

hjetpoluru replied:
@racitores , I am using a mac machine and locally it failed for me when the value is '14.99994' and test passed when the value is '14.99995'. But it passes in the CI strange.
<img width="1392" alt="Screenshot 2025-03-27 at 2 34 55 PM" src="https://github.com/user-attachments/assets/5a5fcb83-6c12-4aaf-9f5e-5f275a87a916" />

seaona replied:
this still seems to be an issue. It continues to fail for me locally, same as Harika. And it also failed on ci in previous runs, meaning, there's a gap of knowledge there, that we need to close.

I wouldn't merge the PR until that's clarified, otherwise, this can start failing in ci anytime this unknown behaviour surfaces :thinking: 

![Screenshot from 2025-03-28 15-22-35](https://github.com/user-attachments/assets/726d8097-d4e5-41cd-b1e0-dd7c8df07585)

seaona replied:
if you don't want to block this PR, I would suggest to maybe comment this line,  create a follow-up ticket and reference the ticket number with a comment in the code :thinking: what do you think?

racitores replied:
https://github.com/MetaMask/metamask-extension/issues/31427

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -75,124 +79,235 @@ async function mockSwapQuotes(mockServer: MockttpServer) {
+      destinationToken: 'WETH',
+      sourceAmount: '10',
+      expectedWethBalance: '60',
+      expectedEthBalance: '14.99992',
```
the spec is failing as this value is incorrect

```suggestion
      expectedEthBalance: '14.99993',
```

racitores replied:
same than before... just wondering if we can ignore this or make it like ~ a value, like balance ~ 14.999

seaona commented on test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts:
```diff
@@ -0,0 +1,297 @@
+import HomePage from '../../page-objects/pages/home/homepage';
+import { NATIVE_TOKEN_SYMBOL, SwapSendPage } from './swap-send-test-utils';
+
+const RECIPIENT_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';
```
nit: if we want to use a variable, we could use the existing one `DEFAULT_FIXTURE_ACCOUNT` from the `constants.ts` file

seaona commented on test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts:
```diff
@@ -0,0 +1,297 @@
+
+describe('Swap-Send ETH', function () {
+  describe('to non-contract address with data that matches swap data signature', function (this: Suite) {
+    it.only('submits a transaction successfully with max amount', async function () {
```
is this needed?

```suggestion
    it('submits a transaction successfully with max amount', async function () {
```

seaona commented on test/e2e/tests/swap-send/swap-send-eth-with-loaded-state.spec.ts:
```diff
@@ -0,0 +1,297 @@
+            '≈ $21.93',
+          );
+
+          // TODO assert swap api request payload
```
q: is this intended to be done in this PR?

racitores replied:
no, it was something copied from the previous / deleted file. Not sure if that make sense anymore. I'll remove it

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -75,124 +79,235 @@ async function mockSwapQuotes(mockServer: MockttpServer) {
-        await assetListPage.clickOnAsset('WETH');
+          // Swap tokens
+          const assetListPage = new AssetListPage(driver);
+          // Remove ETH linea token to avoid potential flakiness
+          if (testCase.sourceToken === 'Ethereum') {
+            await assetListPage.openNetworksFilter();
+            await assetListPage.clickCurrentNetworkOption();
+          }
```
you could avoid this step if you pass it as a fixture

```
.withPreferencesController({
              preferences: {
                tokenNetworkFilter: {
                  '0x1': true,
                }
              }
            })
```

racitores replied:
Thanks for the fixture tip!
I'll try to keep it with the each for now... If it gives any problem in the future will be happy to split it in several tests (adding some functions / flows to reuse the code)

seaona replied:
sounds good, thanks!

seaona replied:
the challenge with the `forEach` approach is that we are not able to perform individual test runs with the `it.only` or we won't be able to skip a specific test if that's needed `it.skip`. If that array grows, this will become quite inconvenient.
Ie: I was running this locally and wanted to run a specific test instance, and I had to comment out one entry, in order to perform the other one.

I would say, this should be avoided as a general practice in our e2e, but no blocker

racitores replied:
I'll have it into account. Lets see how it works in the next weeks and in case it is an inconvenient we will change it quickly 👍

racitores commented on test/e2e/page-objects/pages/swap/swap-page.ts:
```diff
@@ -54,10 +59,32 @@ class SwapPage {
     await this.driver.clickElement(this.swapButton);
-    await this.driver.clickElement(this.closeButton);
+    // console.log('Processing Swap');
+    // await this.swapProcessingMessageCheck('Processing');
```
this is just for testing the pipeline failures. Wonder if we should or not check this. I guess it is a bit dangerous as it doesn't depend on any action

hjetpoluru replied:
@racitores , i think we could have the check for the processing. If yes then could you uncomment it.

racitores replied:
I don't think so, because we don't know how long this is going to take and therefore I think is prone to flakiness (and the value it adds it isn't worthy IMO). If it gets to transaction completed is the same outcome

racitores committed:
fix comments

racitores committed:
remove random value check

racitores committed:
added delays

seaona committed:
fix: flaky test `Settings - general tab validate "मानक हिन्दी" language on tooltips` (#31384)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
In some of the change language specs, we change the language and
immediately click the close settings button. This action can have no
effect, if we click before the language is changes, as then there's a
re-render and we are still in the settings page.

We need to wait until the language has been changed before proceeding
with the next action. For that, we add a check to make sure the language
has changed, every time we perform a `changeLanguage` action.

![Screenshot from 2025-03-28
10-16-42](https://github.com/user-attachments/assets/bbded9ee-533e-4b34-bbdf-e7b05b3fa81f)


ci failure:
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/134139/workflows/c284f602-6903-4a81-a0ac-48a05f987ee7/jobs/4696447/tests

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31384?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Check ci

## **Screenshots/Recordings**
See in the failure artifacts, how after clicking close settings, we
still remain in the settings page. The click was performed but took no
effect, as after that the page was re-rendered with the new language.
Then we try to look for an element from the home page but that can't be
found as we are still in settings



![image](https://github.com/user-attachments/assets/016d8cd0-13a4-4658-8979-e1d599e6a289)


## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

jpuri committed:
fix: cp-12.15.0 support link in EIP-5792 cancel request modal. (#31386)

## **Description**

Fix support link in EIP-5792 cancal request modal.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/4526

## **Manual testing steps**

1. Go to test-dapp
2. Submit EIP-5792 send call request
3. Press cancel button and check learn more link

## **Screenshots/Recordings**
<img width="357" alt="Screenshot 2025-03-28 at 3 27 06 PM"
src="https://github.com/user-attachments/assets/0566ca88-f24d-477c-a920-255af002ea04"
/>

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

jpuri committed:
chore: Updating @metamask/user-operation-controller to version 31.0.0 (#31242)

## **Description**

Updating @metamask/user-operation-controller to version 31.0.0

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/31189

## **Manual testing steps**
NA

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: MetaMask Bot <metamaskbot@users.noreply.github.com>

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -9,9 +9,19 @@ import AdvancedSettings from '../../page-objects/pages/settings/advanced-setting
+  mockEmptyHistoricalPrices,
+  mockEmptyPrices,
+} from '../tokens/utils/mocks';
+import { DEFAULT_FIXTURE_ACCOUNT } from '../../constants';
```
:+1:

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -75,124 +85,234 @@ async function mockSwapQuotes(mockServer: MockttpServer) {
+            .withPreferencesController({
+              preferences: {
+                tokenNetworkFilter: {
+                  '0x1': true,
```
:+1:

micaelae committed:
feat: fetch asset metadata on search (#31258)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This changes adds a `useAssetMetadata` hook that fetches asset details
if it is not found in existing token lists. I've also added more unit
test coverage for the asset-picker token filtering

See [Copilot summary for more
details](https://github.com/MetaMask/metamask-extension/pull/31258#pullrequestreview-2715458493)

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31258?quickstart=1)

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMS-2094

## **Manual testing steps**

1. Select solana as active network
2. Go to Bridge page
3. Get the address of a new pump.fun coin
4. Paste address into the asset picker's search bar
5. The token should appear on the list and be selectable
6. After filling out quote params, quotes should be fetched

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**
No asset is shown
<!-- [screenshots/recordings] -->

### **After**


https://github.com/user-attachments/assets/e1cf7185-49f3-4b64-a31f-1232a19a995d


<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

dan437 committed:
fix: cp-12.16.0 Follow up on the publishBatch hook (#31401)

## **Description**
- This PR is mainly refactoring based on the feedback in the [main
PR](https://github.com/MetaMask/metamask-extension/pull/31267) that was
merged + it updates the STX controller to the latest version.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31401?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Have STX enabled
2. Be on Ethereum mainnet
3. Go to the Send feature
4. You will be able to choose your gas token if you have enough balance
for it and its a supported token

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Signed-off-by: dan437 <80175477+dan437@users.noreply.github.com>

micaelae committed:
fix: balance and token icons are unavailable when the bridge page is reopened (#31343)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31343?quickstart=1)

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMS-2147

## **Manual testing steps**

1. Request a Bridge quote with src=native token
2. Close extension popup
3. Open popup again
4. Balance should be repopulated and token icons should be visible

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

metamaskbot committed:
chore: add workflow_dispatch to security-code-scanner (#30620)

## Summary 

This pull request updates the Security Code Scanner by adding the
`workflow_dispatch` trigger. This trigger allows for the Code Scanner to
be manually triggered from your repository's `Action` tab.


For further questions, please contact the MetaMask Security team.

dbrans committed:
test: fix mocha vs jest type conflicts in test files (#31049)

## **Description**
This PR resolves conflicting types between jest and mocha that were
causing [issues with the type of
"it"](https://github.com/search?q=org%3AMetaMask%20repo%3AMetaMask%2Fmetamask-extension%20it.each%20&type=code)
in unit tests which were masking type errors. It also fixes type issues
that were masked by previously disregarding type errors.


In particular, this PR creates an [isolated TypeScript configuration for
the e2e
tests](https://github.com/MetaMask/metamask-extension/pull/31049/files#diff-bfc1581736f7b790e103572bf8e349eab3346aaa0d29b505f8f9707bfd603e7e)
where they get their own type checking context. Since mocha is only
needed in e2e tests, Mocha types will not be considered for files
outside of that folder.
<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31049?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

GuillaumeRx committed:
revert(snaps): Revert breaking changes from #31058 (#31391)

## **Description**

This reverts the changes made in #31058 because it breaks all the Snaps
confirmations.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31391?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

davidmurdoch committed:
fix: cp-12.15.0 remove sentry cdn url from sentry script (#31360)

We don't use this code path, but just having this url appear in our
extension bundle makes it looks like we could load code from an external
source, something which is not permitted by extension web stores.

To test: download the build (or build it yourself), extract it, search
the source the code for `browser.sentry-cdn.com`. It should not exist.

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.


## **Description**


[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31360?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**


### **Before**


### **After**


## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

-->

seaona committed:
test: rename double and triple ganache references (#31385)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

After this PR was merged:
https://github.com/MetaMask/metamask-extension/pull/30709 all specs are
now running in Anvil by default. In this PR I'm updating more old
ganache references, to make them client agnostic.

There are no functional changes in this PR, so we can skip the quality
gate

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31385?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Check ci -- everything should be working as before

## **Screenshots/Recordings**

n/a

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

georgewrmarshall committed:
chore: updating font family from euclid to centra (#31303)

## Description

This PR updates the default font family across the MetaMask extension
from `Euclid Circular B` to `CentraNo1` to align with brand evolution
updates. The changes include:

1. Updating font family declarations in SCSS and component files from
'Euclid Circular B' to the design tokens
`--font-family-sans`('CentraNo1')
2. Simplifying text component styling by removing brand evolution
feature flag
3. Updating SVG text elements and third party components to use
CentraNo1 font family

The changes are part of the broader [brand evolution
initiative](https://consensys.slack.com/archives/C1L7H42BT/p1742922214475199)
to standardize typography across MetaMask products.

## Related issues

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/4408

## Manual testing steps

1. Build and load the extension
2. Ensure the font files load in the network tab
3. Navigate through various screens and verify text rendering
4. Verify that all text elements are using `CentraNo1` font family
5. Check that no visual regressions have been introduced
6. Pay special attention to:
   - Headers and titles
   - Body text
   - Form inputs

## Screenshots/Recordings

### Before

Navigating around the extension with current Euclid Circular B font
family in expanded and popup view


https://github.com/user-attachments/assets/48cdd000-1762-47d2-90b3-ac37827add48


https://github.com/user-attachments/assets/6422da7e-c0a9-46b6-859e-2f391aa77830

Text component in storybook with curent Euclid Circular B font family


https://github.com/user-attachments/assets/6dc74c95-9857-4bb5-85d7-e610b60e746a

### After
Navigating around the extension with updated CentraNo1 font family in
expanded and popup view


https://github.com/user-attachments/assets/57c06242-5017-4356-8e5f-f7d399404408


https://github.com/user-attachments/assets/f0c8a71b-bc74-4768-a4f3-397c477157b1

Text component in storybook with updated CentraNo1 font family


https://github.com/user-attachments/assets/fee38240-3197-4b01-9a92-849398ab5926

No instances of Euclid left in codebase apart from mock e2e test urls
for `snap-account-abstraction-keyring`

![Screenshot 2025-03-27 at 2 45
51 PM](https://github.com/user-attachments/assets/6fab1f49-e1ac-4466-beae-0b37b548e533)

## Pre-merge author checklist

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md)
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests (updated text component tests)
- [x] I've documented my code (updated SCSS comments and type
definitions)
- [x] I've applied the right labels on the PR

## Pre-merge reviewer checklist

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed)
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots

amandaye0h committed:
style: Update tabs color, hover and animations (#30907)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR modifies the tab colors to improve the visual hierarchy of the
page. It adds a hover effect and animations to make the transitions
smoother.

## **Related issues**

None

## **Manual testing steps**

1. Open MetaMask
2. Hover and click on tabs

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**



https://github.com/user-attachments/assets/77cf3c03-1e07-4633-8df1-c472e9da8f8a


### **After**



https://github.com/user-attachments/assets/3dd38f4d-3c8b-4a24-909b-2aabb8216366


## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Nick Gambino <35090461+gambinish@users.noreply.github.com>

gambinish committed:
fix: token name sort  (#31302)

## **Description**

We recently decided that we wanted to sort by token name, rather than
symbol. The problem is that native tokens for several chains display
`Ethereum` when in actuality the name we get from the token API is
something like `Linea`. This would cause native tokens on some EVM
chains to be miscategorized in sort.

This PR overrides the token name for this scenario. It also consolidates
the `name` and `title` fields to support both EVM and nonEVM chains.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31302?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/31235

## **Manual testing steps**

1. Please test on both flask and non flask builds.
2. EVM and Solana tokens should now be sorted by their display name (not
symbol)
3. Check on both firefox and chrome

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

Before scenarios in bug ticket linked.

### **After**

Chrome:

<img width="360" alt="Screenshot 2025-03-25 at 12 58 45 PM"
src="https://github.com/user-attachments/assets/b5262950-c649-47da-b18c-1b1daa9c5c01"
/>
<img width="1087" alt="Screenshot 2025-03-25 at 1 01 00 PM"
src="https://github.com/user-attachments/assets/a818d72a-22fc-438a-981b-318dc8faedaa"
/>

Firefox:

<img width="539" alt="Screenshot 2025-03-25 at 1 07 11 PM"
src="https://github.com/user-attachments/assets/f6a2143d-c4a3-4d83-b0ca-bc943b6ba77f"
/>


## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

amandaye0h committed:
style: Update color of values to text-default (#30886)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
-->

Updates the color of values from text-alternative to text-default. This
allows users to focus on the information and improves consistency across
pages.

## **Related issues**

None

## **Manual testing steps**

1. Open MetaMask
2. Click on NFT Details

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<img width="355" alt="Screenshot 2025-03-11 at 12 17 22 AM"
src="https://github.com/user-attachments/assets/79db2057-7700-4fdc-8f4a-63b1ee621536"
/>


### **After**

<img width="353" alt="Screenshot 2025-03-11 at 12 19 28 AM"
src="https://github.com/user-attachments/assets/05a3def1-53b9-42d5-9c45-1f622d94ede1"
/>


## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Nick Gambino <35090461+gambinish@users.noreply.github.com>

Gudahtt committed:
fix: cp-12.15.0 Prevent fullscreen UI from opening every startup (#31332)

## **Description**

Prevent tab from opening upon every browser startup.

A [recent PR](https://github.com/MetaMask/metamask-extension/pull/29826)
updated the logic for how we determine whether the extension was just
installed or not; we used to check state, but now we use the
`runtime.onInstalled` event instead to store a flag in session storage
indicating whether the extension was just installed.

Unfortunately this PR had a couple of bugs; the condition using this
flag was accidentally reversed, and the read from session storage was
never successful because of a race condition (the write had not finished
yet).

To address both problems, the logic for detecting first install was
refactored to use a global variable instead of session storage. Globals
can be accessed and written synchronously, preventing any possibility of
a race condition.

The solution was complicated by the fact that MV3 builds must add an
`onInstalled` listener in the root service worker module, otherwise the
listener is never called (we tested this experimentally). It was also
unclear whether the listener would always run before or after the
`background.js` script loading, and we need to trigger an action in
`background.js` on install. But we've accounted for both possibilities
in this solution.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31332?quickstart=1)

## **Related issues**

Fixes: #30924

## **Manual testing steps**

* Create a production-like build (`yarn dist` and `yarn dist:mv2`), or
download the builds from the `metamaskbot` comment
* Test that the window opens only on first install, not on later browser
restarts.

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

dbrans committed:
fix: should not auto-close notification window if loaded in a tab (#30970)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
This PR fixes a rare issue when a user, usually a developer, manually
opens notification.html in a browser tab.

We avoid using `browser.windows.remove(id)` to prevent closing all tabs
in a window.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/30970?quickstart=1)

## **Related issues**

* Fixes: https://github.com/MetaMask/metamask-extension/issues/29821

## **Manual testing steps**
Test that browser tab does not auto-close:
1. Open MM in a browser tab
2. Change url path from /home.html => /notification.html
3. Browser tab should stay open (previously the entire window was
closed)

Test that notification popup auto-closes:
4. Open notification popup by visiting a dapp and doing something (like
connecting) that requires permission
5. Cancel/Decline the request in the notification popup
6. Notification popup should auto-close.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

seaona commented on test/e2e/tests/swaps/swap-erc20-with-loaded-state.spec.ts:
```diff
@@ -75,124 +79,235 @@ async function mockSwapQuotes(mockServer: MockttpServer) {
+          await swapPage.enterSwapAmount(testCase.sourceAmount);
+          await swapPage.selectDestinationToken(testCase.destinationToken);
+          if (testCase.dismissWarning) {
+            await swapPage.dismissManualTokenWarning();
```
I haven't had time to look into this yet. To not block the PR, could we maybe create a ticket and reference it in the code with a comment?

racitores committed:
chore: PR comments addressed

racitores committed:
Merge branch 'main' into add-swap-test-eth-to-weth

racitores committed:
chore: PR comments addressed
